feat: Add custom distributed tracing docs for JS/Node (#8469)
getsentry/sentry-docs: src/platform-includes/distributed-tracing/custom-instrumentation/javascript.mdx

On this page you will learn how to manually propagate trace information into and out of your JavaScript application.

In the following cases, distributed tracing will be set up automatically for you:

- You have our <PlatformLink to="/performance/">performance monitoring feature</PlatformLink> turned on.
- You are using one of the platforms that include distributed tracing out of the box:
  - `@sentry/nextjs`
  - `@sentry/sveltekit`
  - `@sentry/remix`
  - `@sentry/astro`

> If you are using a different package, and have not enabled performance monitoring, you can manually set up your application for distributed tracing to work.

Author: this should go into `node.mdx` only then, this is for browser javascript, right?

> If you are using a different package, and have not enabled performance monitoring, you can manually set up your application for distributed tracing to work.

Author: adding this to the node docs instead 👍

## Enabling Distributed Tracing without Performance

You can add the `BrowserTracing` integration in your `Sentry.init()` options, as described in the Performance <PlatformLink to="/performance/instrumentation/automatic-instrumentation/">Automatic Instrumentation</PlatformLink> docs. If you combine this with `tracesSampleRate: 0`, you will have distributed tracing without performance monitoring:

```javascript
Sentry.init({
  dsn: "___PUBLIC_DSN___",
  integrations: [new Sentry.BrowserTracing()],
  tracesSampleRate: 0,
});
```

## Enabling Distributed Tracing without `BrowserTracing`

> If you do not want to use the `BrowserTracing` integration, you can manually extract and inject tracing information into your application. For this, you must:

Author: hmm not sure, for me "trace your application" is pretty ambiguous (what does this means? sounds like performance monitoring to me?)

Maybe we should overall make this more explicit, like this:

```suggestion
If you do not want to use the `BrowserTracing` integration, you can manually extract and inject tracing data in your application in order to connect multiple systems. For this, you must:
```

or something like this...?

- Extract and store incoming tracing information from HTML `meta` tags when loading the page
- Inject tracing information to the outgoing request when sending outgoing requests.

> To learn more, see our <PlatformLink to="/usage/distributed-tracing/">Distributed Tracing</PlatformLink> docs.

Author: I think the idea of this is that on that page we describe distributed tracing as a concept, generally 🤔  (I copied this from other existing pages for custom instrumentation, FWIW)

On this page you will learn how to manually propagate trace information into and out of your JavaScript application.

In the following cases, distributed tracing will be set up automatically for you:

- You have our <PlatformLink to="/performance/">performance monitoring feature</PlatformLink> turned on.

> - You are using one of the platforms that include distributed tracing out of the box:

Reviewer: ```suggestion
- You're using one of the platforms that include distributed tracing out of the box:
```

On this page you will learn how to manually propagate trace information into and out of your JavaScript application.

Distributed tracing will be set up automatically if:

- You've <PlatformLink to="/performance/">Set Up Performance</PlatformLink> in Sentry.

> - You're using one of the platforms that include distributed tracing out of the box:
>   - `@sentry/nextjs`
>   - `@sentry/sveltekit`
>   - `@sentry/remix`
>   - `@sentry/astro`

Reviewer 2: nit, feel free to ignore

```suggestion
- You're using one of the SDKs that include distributed tracing out of the box:
  - `@sentry/nextjs`
  - `@sentry/sveltekit`
  - `@sentry/remix`
  - `@sentry/astro`
```
